### ISSUE TITLE
Supporting multiple voices, like Host and Guest?

### ISSUE THREAD

**Issue by JT-Apps** [unconv/ai-podcaster#6]
Thank you for your work on this, it's exactly the thing I am looking for for a project...However, I keep only getting one person, the 'host,' talking and then they say introduce the guest (presumably from my .txt description) but there's never another voice that is generated. Transparently I am suing @ProfessorTom's fork as I am on MacOS and their recent changes support that and work but can't seem to figure it out.

Simply put: should this be generating this back and forth dialog between two people?

**Comment by unconv** [comment]
Yes, it should automatically use a different voice for every speaker. To debug, you might want to change the `get_voice` function to be as follows:

```python
def get_voice(name, gender):
    print(f"Getting voice for {name} ({gender})")
    if name not in voices:
        voices[name] = random.choice(voice_names[gender])
        voice_names[gender].remove(voices[name])
    voice = voices[name]
    print(f"Picked voice {voice} for {name}")
    return voice
```

**Comment by JT-Apps** [comment]
> Yes, it should automatically use a different voice for every speaker. To debug, you might want to change the `get_voice` function to be as follows:
> 
> ```python
> def get_voice(name, gender):
>     print(f"Getting voice for {name} ({gender})")
>     if name not in voices:
>         voices[name] = random.choice(voice_names[gender])
>         voice_names[gender].remove(voices[name])
>     voice = voices[name]
>     print(f"Picked voice {voice} for {name}")
>     return voice
> ```

Thank you so much, @unconv! I think I should had to push it a bit further with dialog value (btw, I am unclear on how that functions, quick test shows I can put 1 and get a 21 sec output and I can put 2 and get a 26 sec output...?) and then I got two speakers. 

I hate to make a new thread on this but, after trying for the better part of an hour, any idea how to use one of my custom or created voices? I use the voice.py script/pull from their api and I see the full list, including my custom name and its ID, but I can't seem to modify any of the code to consistently put the main speaker AS a custom ID.  Been tearing my hair out all around [here](https://elevenlabs.io/docs/api-reference/python-text-to-speech-guide#using-cloned-voices) so just thought I'd ask -- thanks!

**Comment by unconv** [comment]
The dialog count means how many back-and-forths of conversation you want in the output. If you set it to one, there will be only one person speaking once. If you set it to two, there will be one person speaking and then one other person. The name "dialog" is a bit out of place here, but I couldn't think of a better word at the time, haha. Maybe would be better to call it "turns".

If you want to use your own custom voices from ElevenLabs, you can just use the custom voice name or the ID (doesn't matter which) in the code. Currently, though, it picks random voices from a list. If you want to use a specific voice, you would have to change the `get_voice` function to return a specific voice for a specific name of speaker. You could do it like this, as a simplistic example:

```python
def get_voice(name, gender):
    if name == "Michael":
        return "michaels-voice-id-or-voice-name"
    if name == "Emily":
        return "emilys-voice-id-or-voice-name"
    return "some-fallback-voice-name-or-id"
```

This of course assumes that you know what name ChatGPT picks for the speaker.

**Comment by unconv** [comment]
@jet3004 BTW, in case you stumbled upon this project without seeing the [video](https://www.youtube.com/watch?v=pSdUPXBzTZ0) where I built it, you might want to check it out to give some clarity on how the code works.
